### PR TITLE
Use default scroll size for analytics `scroll_case_names()`

### PR DESCRIPTION
**Pull request by joxl**
## Technical Summary

The helper function `corehq.apps.reports.analytics.esaccessors.scroll_case_names()` was using a custom scroll size of 10000 (maximum Elasticsearch response size), which isn't a good value to use in this scenario. The function now uses the default scroll size (1000).

Found while researching uses of `ESQuery.scroll()` for #31935.

### Safety story

Default scroll size is safe.

### Automated test coverage

Test added.

### QA Plan

No QA.

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

**Comment by calellowitz** [comment]
> which isn't a good value to use in this scenario

Can you explain more about this? Why is 10000 a poor value? Why is 1000 the best value? Ethan raised a similar question on the original scroll size PR (https://github.com/dimagi/commcare-hq/pull/31931#discussion_r929263302), but I would have also assumed that queries like this that return minimal data and include simple filters on indexed fields would have a larger optimal query size than the complex queries returning the entire documents that use the default value.

**Comment by joxl** [comment]
> Can you explain more about this? Why is 10000 a poor value? Why is 1000 the best value? 

- 10000 is the largest allowable size value (meaning it's the value that is most likely to result in a request timeout, which was the entire goal of #31931).
- 1000 is the existing default
- The [single commit](https://github.com/dimagi/commcare-hq/commit/f6eae0a7fd3a5d6d19b08ef4d5bf439cbf2089bc) that added this scroll does not include any information (e.g. variable names, comments, commit message, tests verifying the use of this value, etc) that suggests the large size is safe or required for performance reasons .

**Comment by joxl** [comment]
Oops, wrong button.

**Comment by esoergel** [comment]
> I would have also assumed that queries like this that return minimal data and include simple filters on indexed fields would have a larger optimal query size than the complex queries returning the entire documents that use the default value.

This is also what I was thinking in the last PR, as Cal mentioned.  Like, I'd expect "ideal page size" to vary depending on some factors like:
* :x: large documents
* :heavy_check_mark: no fields returned (only ID)
* :heavy_check_mark: one or two fields returned
* :x: complex filters (like script or nested queries) or weighted queries
* :heavy_check_mark: only simple, probably cached filters

Also, is the issue page size or the overall duration of the scroll context?  The latter is influenced by HQ processing time in some instances.  As one data point, exports use [this utility](https://github.com/dimagi/commcare-hq/blob/de9f8b4c6e618b25cbe1a0270741592bc1a362cd/corehq/elastic.py#L75-L91) to fully evaluate the scroll query up-front, minimizing the time that context has to live.  It buffers that to disk and re-reads from there during the rest of processing.

Note: That use-case only scrolls over doc ids, so the set of docs comes from the scroll context, but the full doc contents are re-fetched in chunks later, so they might change state mid-processing.  You could imagine pulling the full docs up-front too if needed.

**Comment by joxl** [comment]
These are all great things to discuss. There are a lot of Elasticsearch nuances at play here that are interesting to think about.

What do you think about the safety of this PR?  I'm confident there is no danger in deploying it. Can we move forward with this change as-is?

**Comment by calellowitz** [comment]
> These are all great things to discuss. There are a lot of Elasticsearch nuances at play here that are interesting to think about.
> 
> What do you think about the safety of this PR? I'm confident there is no danger in deploying it. Can we move forward with this change as-is?


After this conversation, I was curious so did a bit of testing and found that iterating over a scroll with a 10K page size was consistently about twice as fast as a 1K page size. That was true on various total result sizes from 5K up to 100k. Since the results are just used to populate a dictionary, mapping the two returned values, there does not appear to be a large risk of the scroll context expiring so I do not believe the performance penalty of this change is not worth it.

In other words, I agree this PR is safe, but I do not think it comes with many benefits, and will cause a performance penalty, so I would not be in favor of merging this.

As an interesting side note, the timings barely changed whether or not the full source was returned, so that part of Ethan and my mental model of ES query timings was incorrect and the size of the returned data does not appear to have a large impact on the query times.
